nav2_smoother/src/nav2_smoother.cpp

```diff
@@ -0,0 +1,298 @@
+                                         "global_costmap/costmap_raw")));
+  declare_parameter("footprint_topic",
+                    rclcpp::ParameterValue(
+                        std::string("global_costmap/published_footprint")));
+  declare_parameter("global_frame", rclcpp::ParameterValue(std::string("map")));
+  declare_parameter("robot_base_frame",
+                    rclcpp::ParameterValue(std::string("base_link")));
+  declare_parameter("transform_tolerance", rclcpp::ParameterValue(0.1));
+  declare_parameter("smoother_plugins", default_ids_);
+
+  // // Launch a thread to run the costmap node
+  // costmap_thread_ = std::make_unique<nav2_util::NodeThread>(costmap_ros_);
+
+  RCLCPP_INFO(get_logger(), "Smoother server created");
+}
+
+SmootherServer::~SmootherServer() {
+  smoothers_.clear();
+  // costmap_thread_.reset();
+}
+
+nav2_util::CallbackReturn
+SmootherServer::on_configure(const rclcpp_lifecycle::State &) {
+  auto node = shared_from_this();
+
+  RCLCPP_INFO(get_logger(), "Configuring controller interface");
+
+  get_parameter("smoother_plugins", smoother_ids_);
+  if (smoother_ids_ == default_ids_) {
+    for (size_t i = 0; i < default_ids_.size(); ++i) {
+      nav2_util::declare_parameter_if_not_declared(
+          node, default_ids_[i] + ".plugin",
+          rclcpp::ParameterValue(default_types_[i]));
+    }
+  }
+
+  tf_ = std::make_shared<tf2_ros::Buffer>(get_clock());
+  auto timer_interface = std::make_shared<tf2_ros::CreateTimerROS>(
+      get_node_base_interface(), get_node_timers_interface());
+  tf_->setCreateTimerInterface(timer_interface);
+  transform_listener_ = std::make_shared<tf2_ros::TransformListener>(*tf_);
+
+  std::string costmap_topic, footprint_topic;
+  double transform_tolerance;
+  this->get_parameter("costmap_topic", costmap_topic);
+  this->get_parameter("footprint_topic", footprint_topic);
+  this->get_parameter("transform_tolerance", transform_tolerance);
+  costmap_sub_ = std::make_shared<nav2_costmap_2d::CostmapSubscriber>(
+      shared_from_this(), costmap_topic);
+  footprint_sub_ = std::make_shared<nav2_costmap_2d::FootprintSubscriber>(
+      shared_from_this(), footprint_topic, 1.0);
+
+  std::string global_frame, robot_base_frame;
+  get_parameter("global_frame", global_frame);
+  get_parameter("robot_base_frame", robot_base_frame);
+  collision_checker_ =
+      std::make_shared<nav2_costmap_2d::CostmapTopicCollisionChecker>(
+          *costmap_sub_, *footprint_sub_, *tf_, this->get_name(), global_frame,
+          robot_base_frame, transform_tolerance);
+
+  if (!loadSmootherPlugins()) {
+    return nav2_util::CallbackReturn::FAILURE;
+  }
+
+  // Initialize pubs & subs
+  plan_publisher_ = create_publisher<nav_msgs::msg::Path>("plan_smoothed", 1);
+
+  // Create the action server that we implement with our smoothPath method
+  action_server_ = std::make_unique<ActionServer>(
+      rclcpp_node_, "smooth_path",
```

Comment:
with set to false, you no longer have the internal node to use here

Reply:
this is a example to create `ActionServer` with dedicated thread but without `rclcpp_node_`.
https://github.com/ros-planning/navigation2/blob/56360c3c45687d09f425aa214d9fe4391e11bfe7/nav2_controller/src/nav2_controller.cpp#L197-L203
you can find more details here:
https://github.com/ros-planning/navigation2/blob/56360c3c45687d09f425aa214d9fe4391e11bfe7/nav2_util/include/nav2_util/simple_action_server.hpp#L52-L69